provider/maas/environprovider.go

```diff
@@ -67,6 +68,24 @@ func (p maasEnvironProvider) CloudSchema() *jsonschema.Schema {
 	return cloudSchema
 }
 
+// Ping tests the connection to the cloud, to verify the endpoint is valid.
+func (p maasEnvironProvider) Ping(endpoint string) error {
+
+	c, err := gomaasapi.NewAnonymousClient(endpoint, apiVersion2)
```

Comment:
I feel that if we do any kind of fallback, it should be at some level above this. Let this know how to talk to a given endpoint; trying multiple endpoints is the responsibility of something else.